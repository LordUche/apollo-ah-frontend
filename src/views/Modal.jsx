/* eslint-disable react/jsx-no-bind */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from 'react';

/**
 * Modal component
 * @param {object} props Component props
 * @param {function} props.passwordResetRequest action creator function
 * @returns {import('@babel/types').JSXAttribute}
 */

function Modal(props) {
  const { clearMessage, title } = props;
  return (
    <div className="modal" id="myModal" onClick={clearMessage}>
      <div className="modal-dialog">
        <div
          className="modal_content"
          onClick={(event) => {
            event.stopPropagation();
            return clearMessage('modalContent');
          }}>
          <div className="cancelTop">
            <button onClick={clearMessage} type="button" className="close" data-dismiss="modal">&times;</button>
          </div>
          <div className="modal_body">
            <div className="modal_header">
              <h4 className="modal_title">{title}</h4>
            </div>
            {props.children}
          </div>
        </div>
      </div>
    </div>
  );
}

export default Modal;

