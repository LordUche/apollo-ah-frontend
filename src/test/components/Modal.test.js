/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Modal from '../../views/Modal';


configure({ adapter: new Adapter() });

describe('<Modal />', () => {
  it('should render <Modal/> component given the props', () => {
    const container = shallow(<Modal />);
    expect(container).toMatchSnapshot();
  });
}); 
