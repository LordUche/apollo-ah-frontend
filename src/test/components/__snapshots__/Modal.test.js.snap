// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render <Modal/> component given the props 1`] = `ShallowWrapper {}`;
